test/services/samples/transfer_service_test.rb

```diff
@@ -76,5 +76,55 @@ def setup
                           name: @current_project.name),
                    exception.result.message
     end
+
+    test 'metadata summary updates after sample transfer' do
+      # Reference group/projects descendants tree:
+      # group12 < subgroup12b (project30 > sample 33)
+      #    |
+      #    ---- < subgroup12a (project29 > sample 32) < subgroup12aa (project31 > sample34 + 35)
+      @sample33 = samples(:sample33)
+      @sample34 = samples(:sample34)
+      @sample35 = samples(:sample35)
+      @project29 = projects(:project29)
+      @project30 = projects(:project30)
+      @project31 = projects(:project31)
+      @group12 = groups(:group_twelve)
+      @subgroup12a = groups(:subgroup_twelve_a)
+      @subgroup12b = groups(:subgroup_twelve_b)
+      @subgroup12aa = groups(:subgroup_twelve_a_a)
+
+      @sample_transfer_params1 = { new_project_id: @project30.id,
+                                   sample_ids: [@sample34.id, @sample35.id] }
+
+      Samples::TransferService.new(@project31, @john_doe).execute(@sample_transfer_params1[:new_project_id],
+                                                                  @sample_transfer_params1[:sample_ids])
+
+      @subgroup12a.reload
+      @subgroup12aa.reload
+      @subgroup12b.reload
+
+      assert_equal({}, @project31.namespace.metadata_summary)
+      assert_equal({}, @subgroup12aa.metadata_summary)
+      assert_equal({ 'metadatafield1' => 1, 'metadatafield2' => 1 }, @subgroup12a.metadata_summary)
+      assert_equal({ 'metadatafield1' => 2, 'metadatafield2' => 2 }, @subgroup12b.metadata_summary)
+      assert_equal({ 'metadatafield1' => 3, 'metadatafield2' => 3 },
+                   @group12.metadata_summary)
+
+      @sample_transfer_params2 = { new_project_id: @project29.id,
+                                   sample_ids: [@sample33.id, @sample34.id, @sample35.id] }
+
```

Comment:
Same comment as above

Reply:
Updated in f76e47740b96955688a5b0d9e0c1c418deacaad4